lib/js/belt_Id.js

```diff
@@ -2,12 +2,48 @@
 
 var Curry = require("./curry.js");
 
+function MakeComparableU(M) {
+  var cmp = M[/* cmp */0];
+  return /* module */[/* cmp */cmp];
+}
+
+function MakeComparable(M) {
+  var cmp = function (a, b) {
+    return Curry._2(M[/* cmp */0], a, b);
```

Comment:
this seems to be a missing optimization. as a work around, can you rewrite the code in this way, it applies below
```ocaml
let cmp = let cmp = M.cmp in fun [@] a b -> cmp a b
```
The take away is we need to avoid a function allocation  here, this matters a lot later